homeassistant/components/snips.py

```diff
@@ -68,12 +74,56 @@ def message_received(topic, payload, qos):
                 slots[slot['slotName']] = {'value': slot['rawValue']}
 
         try:
-            yield from intent.async_handle(
+            intent_response = yield from intent.async_handle(
                 hass, DOMAIN, intent_type, slots, request['input'])
+        except intent.UnknownIntent as err:
+            _LOGGER.warning("Received unknown intent %s",
+                            request['intent']['intentName'])
+            snips_response.add_speech(
+                "This intent is not yet configured within Home Assistant.")
         except intent.IntentError:
             _LOGGER.exception("Error while handling intent: %s.", intent_type)
 
+        if 'plain' in intent_response.speech:
+            snips_response.add_speech(
+                intent_response.speech['plain']['speech'])
+
+        if intent_response.speech:
```

Comment:
This check seems very weird. For example, this means that the error is never sent. 

I am also not 100% convinced that adding a SnipsResponse object is worth it since the data is only ever contained in this function.

```python
speech_response = None

try:
    intent_response = yield from …
    if 'plain' in …:
        speech_response = …
except:
    speech_response = 'Not configured'

if speech_response is None:
    return    # no more work to be done

hass.components.mqtt.async_publish('hermes/dialogueManager/startSession', …)
```

Reply:
@balloob Added a bunch of unittests and cleaned things up a bit. Using the dialog manager to send the notification as the proper way to do it. Also added the snips/duration check per my other snips pr discussion.

The caveat here is that in the unittests I could not get it the unittest to read the mqtt response. Could not find a similar test case where a test listened for an event except for ones where the test  actually fired the event so not sure if this is something that should be working with the mock_mqtt_component or I am doing something wrong. 

I got around it by checking for logging values for the response sent so I think that should be sufficient.